tests/base_scraper/test_base_scraper.py

```diff
@@ -74,5 +82,28 @@ def test_notimplementedexceptions(self):
         self.assertRaises(mozdownload.NotImplementedError,
                           scraper.build_filename, 'invalid binary')
 
+    def test_destination(self):
+        """Test for various destination scenarios"""
+
+        filename = 'download_test.txt'
+        test_url = urljoin(self.wdir, 'download_test.txt')
+
+        #destination is directory
+        scraper = mozdownload.DirectScraper(url=test_url,
+                                            destination=self.temp_dir,
+                                            version=None,
+                                            log_level='ERROR')
+        expected_target = os.path.join(self.temp_dir, filename)
+        self.assertEqual(scraper.target, expected_target)
+
+        #destination is file
+        destination = os.path.join(self.temp_dir, filename)
+        scraper = mozdownload.DirectScraper(url=test_url,
+                                            destination=destination,
+                                            version=None,
+                                            log_level='ERROR')
+        expected_target = destination
```

Comment:
There is no need for this line. Use `destination` directly in the `assertEqual` call.